src/main/java/com/brew/oauth20/server/utils/EncryptionUtils.java

```diff
@@ -1,31 +1,67 @@
 package com.brew.oauth20.server.utils;
 
-import org.apache.tomcat.util.codec.binary.Base64;
-
-import javax.crypto.*;
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.GCMParameterSpec;
 import javax.crypto.spec.SecretKeySpec;
+import java.security.InvalidAlgorithmParameterException;
 import java.security.InvalidKeyException;
 import java.security.NoSuchAlgorithmException;
-
+import java.security.SecureRandom;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
 
 public class EncryptionUtils {
+    private static final String AES_CIPHER_SPEC_KEY = "AES_CIPHER_SPEC";
+    private static final String ALGORITHM_KEY = "ALGORITHM";
+    private static final int IV_SIZE = 12; // Recommended IV size for GCM mode
+    private static final SecureRandom secureRandom = new SecureRandom();
+
     private EncryptionUtils() {
         throw new IllegalStateException("Utility class");
     }
 
-    public static String encrypt(String data, String algorithm, String key) throws NoSuchPaddingException, NoSuchAlgorithmException, InvalidKeyException, IllegalBlockSizeException, BadPaddingException {
-        SecretKey secretKey = new SecretKeySpec(key.getBytes(), algorithm);
-        Cipher cipher = Cipher.getInstance(algorithm);
-        cipher.init(Cipher.ENCRYPT_MODE, secretKey);
+    public static Map<String, String> createAlgorithmKeyHashmap(String algorithm, String cipherSpec) {
```

Comment:
We don't need to take this algorithms as parameter, simply we can define static fields for algorithms.